### PR TITLE
Semaphore

### PR DESCRIPTION
**Pull request by escapewindow**
These are changes to make the chemspill fixes more permanent.
I'm not 100% sure the aiohttp max connections were working as designed. We use semaphores elsewhere (e.g. scriptworker-scripts) and they work very well.

### CODE REVIEW

File: src/scriptworker/utils.py [hunk 1]
```diff
@@ -258,7 +259,9 @@ async def retry_async(
     while True:
         try:
             return await func(*args, **kwargs)
-        except retry_exceptions:
+        except retry_exceptions as exc:
+            if log_exceptions:
+                log.info(f"retry_async exception:\n{type(exc)} {exc}")
```
Comment (JohanLorenzo) [inline src/scriptworker/utils.py:264]: Nit: Should we `log.warning()` instead? I get the feeling an exception is worth being called out as a warning.

File: tests/test_context.py [hunk 2]
```diff
@@ -149,3 +149,12 @@ def test_bad_verify_task(claim_task, bad_path):
...
     with pytest.raises(CoTError):
         context.verify_task()
+
+
+@pytest.mark.asyncio
+async def test_download_semaphore():
+    context = swcontext.Context()
+    sem = context.download_semaphore
+    assert type(sem) == asyncio.BoundedSemaphore
+    assert sem._value == swcontext.DEFAULT_MAX_CONCURRENT_DOWNLOADS
```
Comment (JohanLorenzo) [inline tests/test_context.py:159]: 👍

**Review by JohanLorenzo — APPROVED** [review]
LGTM!

File: src/scriptworker/utils.py (hunk 1 above)
Comment (escapewindow) [reply src/scriptworker/utils.py:264]: Hm. Today, we retry on exception, and we don't log anything if we're retrying; we silently eat the error and sleep, so just logging anything is an improvement.

I don't have a strong objection to warning though, so I can try that.

**Pull request merged**